lib/rubocop/cop/rails/unique_validation_without_index.rb

```diff
@@ -31,7 +31,7 @@ class UniqueValidationWithoutIndex < Base
         RESTRICT_ON_SEND = %i[validates].freeze
 
         def on_send(node)
-          return if uniqueness_part(node).falsey_literal?
+          return if uniqueness_part(node)&.falsey_literal?
```

Comment:
This doesn't fix the entire issue however... 

```
An error occurred while Rails/UniqueValidationWithoutIndex cop was inspecting /Users/mhenrixon/Code/mhenrixon/template/app/models/user.rb:49:2.
undefined method `hash_type?' for nil:NilClass
/Users/mhenrixon/.gem/ruby/3.2.2/gems/rubocop-rails-2.20.0/lib/rubocop/cop/rails/unique_validation_without_index.rb:95:in `column_names_from_scope'
/Users/mhenrixon/.gem/ruby/3.2.2/gems/rubocop-rails-2.20.0/lib/rubocop/cop/rails/unique_validation_without_index.rb:79:in `column_names'
/Users/mhenrixon/.gem/ruby/3.2.2/gems/rubocop-rails-2.20.0/lib/rubocop/cop/rails/unique_validation_without_index.rb:52:in `find_schema_information'
/Users/mhenrixon/.gem/ruby/3.2.2/gems/rubocop-rails-2.20.0/lib/rubocop/cop/rails/unique_validation_without_index.rb:38:in `on_send'
/Users/mhenrixon/.gem/ruby/3.2.2/gems/rubocop-1.52.1/lib/rubocop/cop/commissioner.rb:143:in `public_send'
/Users/mhenrixon/.gem/ruby/3.2.2/gems/rubocop-1.52.1/lib/rubocop/cop/commissioner.rb:143:in `block (2 levels) in trigger_restricted_cops'
/Users/mhenrixon/.gem/ruby/3.2.2/gems/rubocop-1.52.1/lib/rubocop/cop/commissioner.rb:171:in `with_cop_error_handling'
/Users/mhenrixon/.gem/ruby/3.2.2/gems/rubocop-1.52.1/lib/rubocop/cop/commissioner.rb:142:in `block in trigger_restricted_cops'
/Users/mhenrixon/.gem/ruby/3.2.2/gems/rubocop-1.52.1/lib/rubocop/cop/commissioner.rb:141:in `each'
/Users/mhenrixon/.gem/ruby/3.2.2/gems/rubocop-1.52.1/lib/rubocop/cop/commissioner.rb:141:in `trigger_restricted_cops'
/Users/mhenrixon/.gem/ruby/3.2.2/gems/rubocop-1.52.1/lib/rubocop/cop/commissioner.rb:70:in `on_send'
/Users/mhenrixon/.gem/ruby/3.2.2/gems/rubocop-ast-1.29.0/lib/rubocop/ast/traversal.rb:137:in `block in on_dstr'
/Users/mhenrixon/.gem/ruby/3.2.2/gems/rubocop-ast-1.29.0/lib/rubocop/ast/traversal.rb:137:in `each'
/Users/mhenrixon/.gem/ruby/3.2.2/gems/rubocop-ast-1.29.0/lib/rubocop/ast/traversal.rb:137:in `on_dstr'
/Users/mhenrixon/.gem/ruby/3.2.2/gems/rubocop-1.52.1/lib/rubocop/cop/commissioner.rb:71:in `on_begin'
/Users/mhenrixon/.gem/ruby/3.2.2/gems/rubocop-ast-1.29.0/lib/rubocop/ast/traversal.rb:154:in `on_class'
/Users/mhenrixon/.gem/ruby/3.2.2/gems/rubocop-1.52.1/lib/rubocop/cop/commissioner.rb:71:in `on_class'
```

Reply:
Thank you. Can you show us the code to reproduce it?

Reply:
I get it for this line (string, string, integer, if that matters):

```
  validates :key, :name, :terminology_type, presence: true
```

And also here:

```
    validates :contact_email,
      allow_blank: true,
      correct_email: true
```

(correct_email is a custom validator)

This PR fix around half of the errors I had though. Feel free to ask me if a PR fixed the issues, I'd be glad to test it against our codebase

Reply:
This issue will be corrected in the following PR:
- https://github.com/rubocop/rubocop-rails/pull/1028